### PR TITLE
[AppBar] Added a AppBarTypicalCollectionViewExample.

### PR DESCRIPTION
**Pull request by mohammadcazig**
Pivotal: https://www.pivotaltracker.com/story/show/156939716
screen shot 2018-04-25 at 3 01 26 pm
screen shot 2018-04-25 at 3 01 30 pm

### CODE REVIEW

File: components/AppBar/examples/AppBarTypicalCollectionViewExample.m [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+  self = [super initWithCollectionViewLayout:layout];
+
+  if (self) {
+    self.title = @"App Bar";
+
+    // Step 2: Initialize the App Bar and add the headerViewController as a child.
+    _appBar = [[MDCAppBar alloc] init];
+    [self addChildViewController:_appBar.headerViewController];
+
+    MDCSemanticColorScheme *colorScheme = [[MDCSemanticColorScheme alloc] init];
```
Comment (yarneo) [inline components/AppBar/examples/AppBarTypicalCollectionViewExample.m]: please don't theme in init, also please make MDCSemanticColorScheme and MDCTypographyScheme as properties in the interface. see https://github.com/material-components/material-components-ios/pull/3478 for more context

Comment (mohammadcazig) [reply components/AppBar/examples/AppBarTypicalCollectionViewExample.m]: Done.

File: components/AppBar/examples/AppBarTypicalCollectionViewExample.m [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+- (id)init {
+  UICollectionViewFlowLayout *layout = [[UICollectionViewFlowLayout alloc] init];
+  layout.minimumInteritemSpacing = 10.0f;
+
+  self = [super initWithCollectionViewLayout:layout];
+
+  if (self) {
+    self.title = @"App Bar";
+
+    // Step 2: Initialize the App Bar and add the headerViewController as a child.
```
Comment (jverkoey) [inline components/AppBar/examples/AppBarTypicalCollectionViewExample.m:43]: Where is step 1?

**Review by jverkoey — DISMISSED** [review]
Documentation needs to be updated.

File: components/AppBar/examples/AppBarTypicalCollectionViewExample.m (hunk 2 above)
Comment (mohammadcazig) [reply components/AppBar/examples/AppBarTypicalCollectionViewExample.m:43]: added.

**Review by jverkoey — APPROVED** [review]

**Pull request merged**